### ISSUE TITLE
Layouts: Disabling Main page content has no effect

### ISSUE THREAD

**Issue by kswan** [backdrop/backdrop-issues#5808]
## Description of the bug

When managing blocks on a node page layout, if the Main page content is disabled, the content is still displayed on the node page.

## Steps To Reproduce

To reproduce the behavior:
1. Create a page with some content in the body field.
2. Create a layout using the path "node/%".
3. Disable the Main page content block.
4. View the page.

## Actual behavior

The Main page content is displayed even though it is disabled.

## Expected behavior

The disabled block should not be displayed.

## Additional information

Add any other information that could help, such as:
- Backdrop CMS version: 1.23.0

**Comment by laryn** [comment]
Confirmed. (If you fully remove the block it doesn't show, if you disable the block it still shows up).

**Comment by jjmonterey** [comment]
I confirm that I have encountered this behavior also

**Comment by stpaultim** [comment]
We looked at this with @jjmonterey after the weekly meetings this week. 

This seems like it might be a problem that quite a few people would encounter, but not necessarily recognize as a bug. It's pretty easy to work around the problem, but could be very confusing. 

Has anyone had a chance to look at this and see what is causing it and/or if it's an easy or difficult fix?

**Comment by klonos** [comment]
I vaguely recall something about this being by design (@docwilmot or @jenlampton might know more about this). Something about not allowing people to "shoot themselves on the foot".

I think that we should first figure out if there is a legit use case to allow the main content block to be hidden*. In other words, is the issue here an actual bug (are people actually trying to hide the main content block), or simply a matter of inconsistency (the action to disable exists because it is available for all blocks in general, yet does nothing for this specific block). So our approach should be:
  - If the main content block should not be allowed to be disabled/removed from layouts, then we should remove the "Disable" (and "Remove"?) action for this block.
  - If it should be allowed to be disabled/removed (even if edge cases), then we could do one or more of the following things:
    - add a confirmation message before disabling/removing the block, letting people know that this can break things
    - add a warning at the top of the block management UI for layouts where the main content block has been removed/disabled

My quick 2c.

*@kswan, @jjmonterey, @stpaultim can you guys please outline your use case for wanting to disable the main content block?

**Comment by klonos** [comment]
...I just though of a use case: when you want to replace the main content block (which in the case of nodes is one of the display modes with all fields) with individual field blocks managed via the block UI. If that is the most common (or only) use case, then I believe that the reason why we have gotten into this situation is because field blocks were introduced as a feature later (in 1.3.0 with #1100), and most likely we haven't though of this issue here when introducing that.

**Comment by stpaultim** [comment]
@klonos Yes, the use case you described is the one that we had in mind. 

During office hours @jjmonterey demonstrated that after creating a Layout for a type of node and implementing field blocks, he tried to disable the main content block to prevent the layout from displaying two versions of the same node. 

To make this work, he had to "hide" all the fields on the node display mode. Very weird.

**Comment by kiamlaluno** [comment]
Is selecting a different display mode for the layout possible?  
In this way, the layout would use a display mode that does not show the fields already shown in field blocks, but the default display mode would be used in other places where a node is shown.

**Comment by olafgrabienski** [comment]
>> ... on a node page layout, if the Main page content is disabled, the content is still displayed on the node page.

> Confirmed. (If you fully remove the block it doesn't show, if you disable the block it still shows up).

Maybe an oversight when the option to disable the blocks was introduced?

**Comment by klonos** [comment]
> Is selecting a different display mode for the layout possible?

Not possible, but that's what #5774 proposes if I'm not mistaken.

**Comment by jjmonterey** [comment]
Adding a custom display does not override the default display, the default main content block will display even if it is disabled.

![Manage display page](https://user-images.githubusercontent.com/5239865/229568868-fb14f3bc-b90d-4549-8298-bb24c4e11862.png)

![Custom Display mode with main content block enabled](https://user-images.githubusercontent.com/5239865/229570957-52ca3a14-8f96-45e9-a22c-8b98222cc166.png)

 Disabling the main content block does not make a difference

<img width="674" alt="Custom Display mode with main content block disabled" src="https://user-images.githubusercontent.com/5239865/229571267-f1f086cd-fa46-471e-8117-190e8409ebc1.png">

Removing the main content block renders the desired display

<img width="719" alt="Custom Display mode with main content block removed" src="https://user-images.githubusercontent.com/5239865/229571594-bd3315f4-c542-4d18-864d-798820e82c63.png">

Disabling the main content block AND hiding the fields in the default display also renders the desired layout

<img width="680" alt="Default display fields hidden" src="https://user-images.githubusercontent.com/5239865/229572131-068ac6e5-9d7d-46bf-8b6e-6f3a221f468b.png">

<img width="666" alt="Main content block disabled but all fields hidden" src="https://user-images.githubusercontent.com/5239865/229572201-b687f665-0e22-4f8f-9bb6-cc3404312399.png">

In Drupal using Display Suite layouts Display Suite creates a custom layout in the CONTENT Section only although other custom regions can be displayed or disabled.

![Drupal display](https://user-images.githubusercontent.com/5239865/229574408-fb56502c-83e9-4443-a763-d97077319824.png)

![Drupal HTML Display Suite](https://user-images.githubusercontent.com/5239865/229575980-cf277c98-9783-4d43-a173-e824ad8cd2f4.png)

The advantage of  being able to render a custom layout inside the content region is that a Boxton Layout could be used almost exclusively.

It seems to me that selecting a custom Display Mode should immediately bring up a configure layout screen to add a custom layout, since creating a custom display mode will use the default layout if it doesn't specify a custom layout - somewhat defeating the purpose of a custom display mode.

**Comment by kswan** [comment]
@klonos:
> ...I just though of a use case: when you want to replace the main content block in `node/%` layouts (in which case it is actually one of the display modes with all its fields) with individual field blocks managed via the block UI. If that is the most common (or only) use case, then I believe that the reason why we have gotten into this situation is because field blocks were introduced as a feature later (in 1.3.0 with #1100), and most likely we haven't though of this issue here when introducing that.

Yes, this is the use case that resulted in this issue.

A contributing factor is the note shown on the Main page content block that states "This block may be required for this page to work properly." Due to this note, I was hesitant to remove the block and thought it safer to just disable it. 
![image](https://user-images.githubusercontent.com/154391/229605399-c95c0a98-5e9f-46bb-8aa4-12d654866ac1.png)

Since disabling the block had no effect, I removed the block and the layout works fine.
